### PR TITLE
Submit game AAAAXY (io.github.divverent.aaaaxy)

### PR DESCRIPTION
**Pull request by divVerent**
Note that the AppStream data is in the upstream repo itself:

https://github.com/divVerent/aaaaxy/blob/main/io.github.divverent.aaaaxy.metainfo.xml

### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
  - IPC, X11, DRI: used for using OpenGL access of the game
  - Pulseaudio: used for sound playback of the game
  - --device=all: used to access gamepads
- [ ] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/divVerent/aaaaxy (i.e. I _am_ upstream)
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*
  - modules.txt is output of "go mod vendor" in the upstream repo, and is necessary to be able to build the project without internet access. In theory it can be derived from the go.mod file, but I am not aware of any tools to do that. I had tried using https://github.com/flatpak/flatpak-builder-tools/tree/master/go-get instead, but my project does not build in GO111MODULES=off mode and I see no avenue to make it so, so I am using this "go vendor" based approach instead.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: io.github.divverent.aaaaxy.yml [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.golang
+command: aaaaxy
+cleanup:
+  - /builddeps
+finish-args:
+  # Needed to access gamepads.
+  - --device=all
+  # Needed for 3D graphics.
+  - --device=dri
```
Comment (hfiguiere) [inline io.github.divverent.aaaaxy.yml:15]:
```suggestion
```

It's part of --device=all

**Comment by divVerent** [comment]
As for the other errors:

```
builddir/files/share/appdata/io.github.divverent.aaaaxy.appdata.xml: FAILED:
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
• tag-missing           : <release> required
• tag-missing           : <url> is not present
```

Is it OK to initially NOT tag this upstream? For ever upstream tag I need to make a new binary release for all others; so I'd prefer if I could just point at a commit that is today's release plus this change.

**Comment by hfiguiere** [comment]
You can just add a patch. Fix it upstream and bring that patch to the flathub build.

**Comment by divVerent** [comment]
Anything other than <release> is easy to add. I really don't want to maintain all releases in yet another way, so if there's no way to provide this from GitHub Releases automatedly, I'd rather want to opt out of that.

Now trying to find ways around that (e.g. just linking to latest on Github somehow).

**Comment by divVerent** [comment]
Actually, it's outright impossible to have the correct `<release>` in an upstream file, as it requires having checksums of the binaries built from this, which _contain_ this very file. This is... impossible without breaking the hash.

So for now I'll just link to https://github.com/divVerent/aaaaxy/releases/latest and put today's date; this is not ideal but I see no better way right now.

**Comment by divVerent** [comment]
Let's do this; I created a new tag that the rest of my stuff should ignore.

**Comment by divVerent** [comment]
Can you try a new test build? This should all be fixed now.

**Comment by hfiguiere** [comment]
you don't need specific permission. you can request it yourself.

**Comment by barthalion** [comment]
bot, build io.github.divverent.aaaaxy

**Comment by barthalion** [comment]
bot, build io.github.divverent.aaaaxy

**Comment by divVerent** [comment]
My attempt at trying to fix - by adding an `<icon>` to the metainfo xml file - ended in "tag-invalid : not allowed in desktop appdata".

So how _should_ one fix this error?

**Comment by divVerent** [comment]
I think I found it - had a wrong launchable in the appstream file.

**Comment by divVerent** [comment]
bot, build io.github.divverent.aaaaxy

**Comment by divVerent** [comment]
bwrap: Can't get type of source /srv/buildbot/.cache/.flatpak/3344334271: No such file or directory

That's odd, and only happened on one arch. I'll retry.

**Comment by divVerent** [comment]
bot, build io.github.divverent.aaaaxy

**Comment by divVerent** [comment]
Trying on a fresh Fedora install now (I built/tested the Flatpak on Debian). Works fine; one issue though: software center shows no screenshots, but "forbidden signs" where screenshots should go.

Let me see what that is.

**Comment by barthalion** [comment]
This is actually expected. We mirror screeenshots and rewrite URLs to use our CDN, but it's not supported for test builds like this.

**Comment by divVerent** [comment]
Sounds good then, thanks!

**Comment by barthalion** [comment]
(It kinda worked, but not fully. Let's try again as I pushed out some fixes to the automation.)

/merge

**Pull request closed without merging**